### PR TITLE
Mask-based nullable columns for CTable

### PR DESCRIPTION
**Pull request by FrancescAlted**
A nullable CTable column now keeps its nulls in a per-column .notnull validity sidecar — Arrow's own model — instead of reserving a value from its own range, and that is what a bare nullable=True resolves to.

A sentinel is lossy by construction: a nullable int8 could not hold -128, free-text utf8 had no safe sentinel at all, bool(nullable=True) silently became uint8 with a reserved 255, and Arrow columns whose type had no value to spare could not be imported. With a side channel, to_arrow(from_arrow(x)) returns x for nullable bool, full-range int8/uint8, float64 holding nan/±inf/-0.0 as values, utf8 holding "" and "__BLOSC2_NULL__", and timestamp holding int64.min — none of which survive a sentinel. complex becomes nullable at all.

Sentinel storage is supported indefinitely and is one keyword away (null_storage="sentinel", any explicit null_value=, or a NullPolicy field). Nothing on disk changes: opening a stored table never re-resolves anything, so every existing table keeps the storage, dtype and sentinel it was written with. A table containing a mask column records schema version 3, which older readers refuse with a clear error rather than misreading. convert_nulls() moves columns between the two in either direction, never implicitly.

Landed in ten phases, each independently reviewable, ending with null-aware column indexes: per-segment extrema are taken over the rows that carry a value, so min/max answer from the index for a nullable column (236x on 20M rows) and where() with an OR over one uses the index instead of scanning.

Ten bugs were found and fixed on the way that have nothing to do with mask storage and are reachable today — descending sort_by on a bool column raised and on full-range signed ints ordered wrongly; group_by min over a bool column always returned False; add_column() after copy() backfilled one row short; and t.where("a > 10") returned a nullable column's nulls as matches on both the scan and the index paths.

plans/mask-based-nulls.md is the reference document for the design and carries the as-built record, including nine premises the implementation disproved. ~350 new tests across seven files; full suite green (8765 passed).

### CODE REVIEW

File: src/blosc2/ctable.py [hunk 1]
```diff
@@ -8606,14 +9315,24 @@ def to_csv(self, path: str | None = None, *, header: bool = True, sep: str = ","
...
                         json_strings.append(json.dumps(arr[i].tolist()))
                 arrays.append(json_strings)
+            elif col.null_storage == NULL_MASK and col.null_count():
+                # An empty field is CSV for missing, and for a mask column it is
+                # the *only* way to say it -- writing the fill would come back as
+                # a real 0 or "".  A sentinel column keeps writing its sentinel,
+                # which from_csv maps back, so its output is unchanged.
+                arrays.append(
+                    ["" if is_null else v for v, is_null in zip(col[:], col.is_null(), strict=True)]
+                )
```
[the remark this answers is not in the document]
Comment (FrancescAlted) [reply src/blosc2/ctable.py:9843]: Good catch — fixed in 6c04dca4.

A mask-backed **text** column no longer writes nulls as empty fields, since that field cannot mean both `""` and missing. It now writes `\N` for a null and `\E` for a valid `""`, and escapes any value already shaped like one (`\N` -> `\\N`, `\\N` -> `\\\N`); `from_csv` inverts it. Numeric/bool mask columns keep the empty field (no ambiguity there), and sentinel columns write their sentinel exactly as before, so existing CSV output is unchanged.

Reading text also stops calling `strip()` before testing for an empty field, so whitespace-only text stays a real value; the strip tolerance remains for numeric columns, where a padded field is just whitespace.

`test_csv_mask_text_roundtrip_empty_vs_null` round-trips `["", "  ", None, "ok", "\N", "\E", "\\N", "N"]` through `to_csv`/`from_csv` and checks both the values and `is_null()`.

(`csv.QUOTE_NOTNULL` would have been the stdlib answer, but it needs Python 3.12 and the package targets >= 3.11.)
